presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestDecimalAverageAggregation.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.LongDecimalWithOverflowAndLongState;
+import com.facebook.presto.operator.aggregation.state.LongDecimalWithOverflowAndLongStateFactory;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.DecimalType;
+import com.facebook.presto.spi.type.UnscaledDecimal128Arithmetic;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static com.facebook.presto.operator.aggregation.DecimalAverageAggregation.average;
+import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
+import static com.facebook.presto.spi.type.UnscaledDecimal128Arithmetic.unscaledDecimal;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestDecimalAverageAggregation
```

Comment:
>  Nevertheless to which benchmark does your comment apply? Here you commented unit tests.

I think it was a general comment, but I had to leave it there since github doesn't allow per-commit comments.